docs/source/extending/overloading-guide.rst

```diff
@@ -0,0 +1,251 @@
+is about testing if the function works correctly. Most of the test code in this
+example checks that all error cases are handled and that all raised exceptions
+are of the correct type and have the correct error message. When implementing
+tests cases for Numba, you should always use ``numba.tests.support.TestCase``.
+
+.. literalinclude:: test_myjitmodule2.py
+
+While is is a pretty descent test, it wouldn't be accepted into the Numba.
+There are a few more test-cases that should be implemented:
+
+* Empty array
+* Single value array
+* Multidimensional arrays
+* Tests for ``int32`` and ``float32`` types
+
+You can download the example code from above including tests here:
+
+* :download:`mymodule.py </extending/mymodule.py>`
+* :download:`myjitmodule.py </extending/myjitmodule.py>`
+* :download:`myjitmodule2.py </extending/myjitmodule2.py>`
+* :download:`test_myjitmodule2.py </extending/test_myjitmodule2.py>`
+
+Implementing ``overload`` for Numpy functions
+=============================================
+
+Numba has strong support for NumPy because it provides ``jit`` compatible
+re-implementations of NumPy functions. In such cases ``overload`` is a very
+convenient option, however there are a few additional things to watch out for.
+
+* The  Numba implementation should match the NumPy implementation as closely as
+  feasible with respect to accepted types, arguments, raised exceptions and
+  runtime (Big-O / Landau order).
+
+* If you are implementing a new function, you should always update the
+  `documentation
+  <http://numba.pydata.org/numba-doc/latest/reference/numpysupported.html>`_.
+  The sources can be found in `docs/source/reference/numpysupported.rst``. Be
+  sure to mention any limitations that your implementation has, e.g. no support
+  for the ``axis`` keyword.
+
+* When writing tests for exceptions, for example, when adding tests to
+  ``numba/tests/test_np_functions.py`` you may encounter the following error
+  message:
+
+  .. code:: pycon
+
+        ======================================================================
+        FAIL: test_foo (numba.tests.test_np_functions.TestNPFunctions)
+        ----------------------------------------------------------------------
+        Traceback (most recent call last):
+        File "/Users/vhaenel/git/numba/numba/tests/support.py", line 645, in tearDown
+            self.memory_leak_teardown()
+        File "/Users/vhaenel/git/numba/numba/tests/support.py", line 619, in memory_leak_teardown
+            self.assert_no_memory_leak()
+        File "/Users/vhaenel/git/numba/numba/tests/support.py", line 628, in assert_no_memory_leak
+            self.assertEqual(total_alloc, total_free)
+        AssertionError: 36 != 35
+
+  This is caused because some exceptions leak references. Ideally, you will
+  place all exception testing in a separate test method and then add a call to
+  ``self.disable_leak_check()`` to disable the leak-check.
```

Comment:
True - maybe also mention that your test class will need to inherit the relevant mixin to allow you to call and use disable_leak_check.

Reply:
This was mentioned above, do you think I should repeat myself here?

Reply:
So, the mention is earlier in the document, I added a reminder here.